rcl/src/rcl/node.c

```diff
@@ -284,15 +284,6 @@ rcl_node_init(
     // error message already set
     goto fail;
   }
-  // The initialization for the rosout publisher requires the node to be in initialized to a point
-  // that it can create new topic publishers
-  if (rcl_logging_rosout_enabled() && node->impl->options.enable_rosout) {
-    ret = rcl_logging_rosout_init_publisher_for_node(node);
```

Comment:
It feels to me that if we are going to make the higher-layer library call `rcl_logging_rosout_init_publisher_for_node`, then we should also make the higher-layer library call `rcl_logging_rosout_fini_publisher_for_node`.  That is, we should remove that call from `rcl_node_fini` as well.

Reply:
addressed with https://github.com/ros2/rcl/pull/1065/commits/70b801a8bec97a80a6f316394f72e4119db62e1e